### ISSUE TITLE
Floating IP acceptance test failing

### ISSUE THREAD

**Issue by dbaumgarten** [gophercloud/gophercloud#272]
Hi,

I just noticed that the acceptance-tests for floating-ips aren't working.

Running: go test -v floatingips_test.go layer3.go  (in the appropriate folder)
Results in:
` networking.go:83: Attempting to create subnet: TESTACC-RXoeMDT0
        networking.go:90: Successfully created subnet.
        layer3.go:46: Attempting to create external router: TESTACC-QXGreeZZ
        floatingips_test.go:55: Unable to create router: Invalid request due to incorrect syntax or missing required parameters.
        networking.go:188: Attempting to delete subnet: 080b375b-b3aa-413f-b356-c3beb692483a
        networking.go:195: Deleted subnet: 080b375b-b3aa-413f-b356-c3beb692483a`

And so the Test fails.

I am testing on OpenStack Liberty.

**Comment by jtopjian** [comment]
@dbaumgarten The floating IP test was indeed broke. #274 fixes it.

The README at https://github.com/gophercloud/gophercloud/tree/master/acceptance mentions how to run acceptance tests. For the floating IP one, I did:

```shell
$ gophercloudtest TestLayer3FloatingIPsCreateDelete networking/v2/extensions/layer3
```

I also had to be an admin user since it's creating a subnet on the default Neutron network in my environment. That might vary with your environment.

Let me know if you're still having issues. :)

**Comment by dbaumgarten** [comment]
Ok. I looked at this at little further.
The tests failed, because the tried to add a subnet to my floating-ip-network. That didn't work, because the used account doesnot have admin privileges (thank god).

Why are the tests doing this? Why aren't they creating their subnets on a "normal" network?
Adding a subnet to the floating-ip network is nothing I would like a test to do on my productive OpenStack environment. And I also would not want the test to run with admin privileges. (If not absolutely necessary)

Can't we just make the tests use a normal network? What about using the network provided via OS_NETWORK_NAME ? We could just look up its id and use this for the subnets.

**Comment by jtopjian** [comment]
That indeed looks a bit strange. I want to say that's a mistake, but something is making me think there was a reason why I made the test that way. It might have been because the ability to create a router with interfaces on the non-admin subnet and the public subnet was not possible at the time.

If you're able to, feel free to rewrite the test to account for a more normal workflow. I can also take a look when I get a moment, too, but it might take me a few days.

**Comment by jrperritt** [comment]
closed by #277
